include/turtle/matcher.hpp

```diff
@@ -39,7 +39,12 @@ class matcher<const char*, const char*>
 {
 public:
     explicit matcher(const char* expected) : expected_(expected) {}
-    bool operator()(const char* actual) { return std::strcmp(actual, expected_) == 0; }
+    bool operator()(const char* actual)
+    {
+        if (nullptr == actual || nullptr == expected_)
```

Comment:
Would it not be more generic if it were
```suggestion
        if (!actual || !expected_)
```
?